libraries/AdaptiveExpressions/Expression.cs

```diff
@@ -527,10 +530,16 @@ public void ValidateTree()
         /// <see cref="System.Collections.IDictionary"/> otherwise reflection is used to access property and then indexer.
         /// </param>
         /// <param name="options">Options used in the evaluation. </param>
+        /// <param name="locale">locale used in the evaluation. </param>
         /// <returns>Computed value and an error string.  If the string is non-null, then there was an evaluation error.</returns>
-        public (T value, string error) TryEvaluate<T>(IMemory state, Options options = null)
+        public (T value, string error) TryEvaluate<T>(IMemory state, Options options = null, string locale = null)
         {
             var opts = options ?? new Options();
+            if (locale != null)
+            {
+                Thread.SetData(Thread.GetNamedDataSlot("locale"), locale);
```

Comment:
The other question is for storage should we use Thread.GetNamedDataSlot or we should use thread.currentThread.culture? 

cc @tomlm @chrimc62

Reply:
now stored in Thread.CurrentThread.CurrentCulture